asserts/model.go

```diff
@@ -523,6 +523,14 @@ func (mod *Model) StorageSafety() StorageSafety {
 	return mod.storageSafety
 }
 
+// SnapdSnap returns the details of the snapd snap for this model.
+func (mod *Model) SnapdSnap() *ModelSnap {
+	if mod.Base() == "" || mod.Base() == "core" {
```

Comment:
A comment explaining why these checks are made might be useful. Additionally, rather than directly accessing the slice of snaps, maybe store a pointer to the snap on `*Model` like the rest of the essential snaps. Although that really doesn't matter too much.

Reply:
I think core18 style classic models can have no base but a snapd snap?

Reply:
But then how you would know whether to seed core or core18? I don't see examples of 18.04 classic models in https://github.com/snapcore/models/tree/master , do you know where could I find one?

Reply:
I think the hybrid installer might have one, maybe some cloud images. We should have some example in our own unit test somewhere, though they test might be signing it on the fly.

Reply:
also on classic afair if there's a snapd it will be used in preference to core

Reply:
I've changed the strategy and now I simply check the name of the first snap in the slice, to avoid this issues. The only example that I've seen of classic 18 model is `snapd/tests/lib/assertions/developer1-my-classic-w-gadget-18.model`, which has:
```
required-snaps:
  - core18
```